swupd/helpers_test.go

```diff
@@ -30,6 +30,11 @@ func mustDirExistsWithPerm(t *testing.T, path string, perm os.FileMode) {
 }
 
 func mustSetupTestDir(t *testing.T, testName string) string {
+	oldDirs, _ := filepath.Glob("./testdata/cmtest-" + testName + "*")
+	for _, d := range oldDirs {
+		removeAllIgnoreErr(d)
+	}
+
 	testDir, err := ioutil.TempDir("./testdata", "cmtest-"+testName)
```

Comment:
and change this to

    testDir, err := ioutil.TempDir("./testdata", "cmtest-"+testName +".")

to avoid the issue (at least until people start adding their of periods to test names).
Obviously one can use a different character than `.`, any of `!@#%+` are reasonable choices, or even adding `@@`

Reply:
Good suggestion, thank you.